pkg/controller/vsphere/reconciler_test.go

```diff
@@ -762,6 +762,92 @@ func TestReconcileTags(t *testing.T) {
 	}
 }
 
+func TestCheckAttachedTag(t *testing.T) {
+	model, session, server := initSimulator(t)
+	defer model.Remove()
+	defer server.Close()
+
+	managedObj := simulator.Map.Any("VirtualMachine").(*simulator.VirtualMachine)
+	managedObjRef := object.NewVirtualMachine(session.Client.Client, managedObj.Reference()).Reference()
+
+	vm := &virtualMachine{
+		Context: context.TODO(),
+		Obj:     object.NewVirtualMachine(session.Client.Client, managedObjRef),
+		Ref:     managedObjRef,
+	}
+
+	tagName := "CLUSTERID"
+
+	if err := session.WithRestClient(context.TODO(), func(c *rest.Client) error {
+		tagsMgr := tags.NewManager(c)
+
+		id, err := tagsMgr.CreateCategory(context.TODO(), &tags.Category{
+			AssociableTypes: []string{"VirtualMachine"},
+			Cardinality:     "SINGLE",
+			Name:            "CLUSTERID_CATEGORY",
+		})
+		if err != nil {
+			return err
+		}
+
+		_, err = tagsMgr.CreateTag(context.TODO(), &tags.Tag{
+			CategoryID: id,
+			Name:       tagName,
+		})
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}); err != nil {
+		t.Fatal(err)
+	}
+
+	testCases := []struct {
+		name    string
+		findTag bool
+	}{
+		{
+			name:    "Successfully find a tag",
+			findTag: true,
+		},
+		{
+			name: "Fail to find a tag",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			if err := session.WithRestClient(context.TODO(), func(c *rest.Client) error {
```

Comment:
I wonder if this testcase could be simpler if we attach a tag outside of the testcases and then within the testCase, provide a tag name to check if it's attached, intstead of `findTag`, this will mean inside each test case all you need to do is call `checkAttachedTag` and the output of that method

I think this way we have all of the setup done outside of the individual test cases and then exercise our test code in each case rather than also having some of the setup and teardown in each test case, WDYT?

Reply:
fixed, thanks